R/FilterStateChoices.R

```diff
@@ -92,6 +92,9 @@ ChoicesFilterState <- R6::R6Class( # nolint
     #' optional `is.na(<varname>)`.
     #' @return (`call`)
     get_call = function() {
+
+      if (private$is_disabled()) return(NULL)
```

Comment:
as state is cached the call will return NULL anyway - this `if` is unnecessary

Reply:
So I made this change and now dates and datetimes are throwing errors. More specifically, the [type check](https://github.com/insightsengineering/teal.slice/blob/c6fec4c6bf946a5c72e9c7367268640d3de97754/R/call_utils.R#L212) in `call_condition_range_date` is failing.

There's some odd reactivity behavior going on here. When a filter state is disabled (and state set to `NULL`) **only** the date and datetime `get_call` methods are called. This is where the error occurs. For the other filter state types though, `get_call` is _not_ called on disabling. Any idea what might be causing this? Should I revert this change for dates and datetimes?

Reply:
Never mind, I found it.